### ISSUE TITLE
Cognito Sign In asks to confirm sign in as previously signed in user

### ISSUE THREAD

**Issue by lherna1371** [mheffner/rails-cognito-example#1]
Not really an issue, but I'm wondering how to force Cognito to require login credentials upon each login request.  Using your repository I am forcing a logout prior to the Cognito 1 hour token expiration. This works fine as the cognito_token is deleted. However, when I attempt to sign in via sign in link and the Cognito UI I am presented with a form to either sign in using the previous user just logged out or to click on a link to sign in as another user. Clicking on the button to sign in as a the previous user logs me in without requesting a password or MFA, this is an issue. I am wondering is this is a setting on the Cognito side of things as I can delete the sessions and cognito_sessions and I still encounter this problem. Any help in implementing this will be really helpful. Ultimately my goal is to simply force  a signout and require full Cognito authentication once again. Thanks.

**Comment by mheffner** [comment]
Hi @lherna1371 ,

So how are you "forcing a logout prior to the Cognito 1 hour token expiration"? I think to force Cognito to drop its session token, and hence force a new login, you need to redirect the user to the [logout endpoint](https://docs.aws.amazon.com/cognito/latest/developerguide/logout-endpoint.html).

Could you detect a user that *should* be logged out prior to the 1 hour expiration and the next time they hit your app, force redirect them to that logout endpoint? That might kill their cognito session so they have to input password/MFA again. The "logout and prompt sign in" URI on that doc page might be a better fit in that scenario.

**Comment by lherna1371** [comment]
Hi. As you suggested, I ended up doing a redirect to the logout endpoint. I just checked a user's cognito session expiration time. Created an instance variable to define a 15 min session length and checked the cognito session's expiration time against it in a login_check before action. Essentially what the example you provided is doing, just lowered the session duration and I do not do a token refresh. I will implement a token refresh prompt/warning so that users can refresh it or else they are forced to logout if they ignore it. All is working as expected now, thank you for your response.
